server/app/repository/ReportingRepository.java

```diff
@@ -42,15 +47,24 @@ public ImmutableList<ApplicationSubmissionsStat> loadMonthlyReportingView() {
         .findList()
         .stream()
         .map(
-            row ->
-                ApplicationSubmissionsStat.create(
-                    row.getString("program_name"),
-                    Optional.of(row.getTimestamp("submit_month")),
-                    row.getLong("count"),
-                    getSecondsFromPgIntervalRowValue(row, "p25"),
-                    getSecondsFromPgIntervalRowValue(row, "p50"),
-                    getSecondsFromPgIntervalRowValue(row, "p75"),
-                    getSecondsFromPgIntervalRowValue(row, "p99")))
+            row -> {
+              String programName = row.getString("program_name");
+              return ApplicationSubmissionsStat.create(
+                  versionRepositoryProvider
+                      .get()
+                      .getProgramByNameForVersion(programName, activeVersion)
+                      .get()
+                      .getProgramDefinition()
+                      .localizedName()
+                      .getDefault(),
```

Comment:
issue (blocking): This makes an additional database query for each row in the result set, i.e. it's an [N+1 query](https://schneide.blog/2021/12/06/understanding-identifying-and-fixing-the-n1-query-problem/). The reporting view and repository are designed to load all necessary data for the reporting pages in a single query to reduce database load. Please adjust your approach here to maintain that behavior.